sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -283,6 +292,49 @@ public void close()
     planner.close();
   }
 
+  private SqlNode processStatementList(SqlNode root, String sql)
+  {
+    if (root instanceof SqlNodeList) {
+      final SqlNodeList nodeList = (SqlNodeList) root;
+      if (!allowSetStatementsToBuildContext && nodeList.size() > 1) {
+        throw InvalidSqlInput.exception("Multiple statements detected in SQL string[%s], but only a single statement is supported",
+                                        sql
+        );
+      }
+      final Map<String, Object> contextMap = new LinkedHashMap<>();
+      boolean isMissingDruidStatementNode = true;
+      // convert 0 or more SET statements into a Map of stuff to add to the query context
+      for (int i = 0; i < nodeList.size(); i++) {
+        SqlNode sqlNode = nodeList.get(i);
+        if (sqlNode instanceof SqlSetOption) {
+          final SqlSetOption sqlSetOption = (SqlSetOption) sqlNode;
+          if (!(sqlSetOption.getValue() instanceof SqlLiteral)) {
+            throw InvalidSqlInput.exception("Invalid sql SET statement[%s], value must be a literal", sqlSetOption);
```

Comment:
What does the interpolated `sqlSetOption` end up looking like? I'm just wondering if the error messages are going to be "nice".

SQL should be capitalized, btw.

I also wonder if it might be nicer to interpolate just the name of the parameter.

Reply:
switched to use toSqlString - I think its ok with the whole statement, but can switch to just identifier if you think it would be better. this and all other error messages can be seen in tests https://github.com/apache/druid/pull/17894/files#diff-c8b6119838f7153eb393d6c42f5be66179b47a2819494e7319afd962f4d2b333R15933